File: app/src/main/java/com/jmalinen/blackjack/model/ChartCell.kt
package com.jmalinen.blackjack.model

enum class ChartCell(val symbol: String) {
    HIT("H"),
    STAND("-"),
    DOUBLE_HIT("D"),
    DOUBLE_STAND("D/S"),
    SPLIT("Y"),
    SPLIT_HIT("Y/H"),
    SURRENDER_HIT("Rh"),
    SURRENDER_STAND("Rs"),
    SURRENDER_SPLIT("Ry"),
}

data class ChartRow(
    val label: String,
    val cells: List<ChartCell>
)

data class StrategyChartData(
    val hardRows: List<ChartRow>,
    val softRows: List<ChartRow>,
    val pairRows: List<ChartRow>,
)

File: ios/Blackjack/Blackjack/Model/ChartCell.swift
enum ChartCell {
    case hit
    case stand
    case doubleHit
    case doubleStand
    case split
    case splitHit
    case surrenderHit
    case surrenderStand
    case surrenderSplit

    var symbol: String {
        switch self {
        case .hit: "H"
        case .stand: "-"
        case .doubleHit: "D"
        case .doubleStand: "D/S"
        case .split: "Y"
        case .splitHit: "Y/H"
        case .surrenderHit: "Rh"
        case .surrenderStand: "Rs"
        case .surrenderSplit: "Ry"
        }
    }
}

struct ChartRow: Equatable {
    let label: String
    let cells: [ChartCell]
}

struct StrategyChartData: Equatable {
    let hardRows: [ChartRow]
    let softRows: [ChartRow]
    let pairRows: [ChartRow]
}
